### PR TITLE
Add CRR E2E tests

### PR DESCRIPTION
**Pull request by welansari**
Issue: ZENKO-5069

### CODE REVIEW

File: tests/zenko_tests/e2e_config/accounts.py [hunk 1]
```diff
@@ -50,7 +51,12 @@ def create_account_secret(name, credentials, namespace="default"):
                 "type": "end2end",
             },
         ),
-        string_data=credentials,
+        string_data={
```
Comment (francoisferrand) [inline tests/zenko_tests/e2e_config/accounts.py:54]: these secret are only used to store the creds from the "config" python script to the "test runner" script:
- may be read directly by the test
- or better yet, may be setup directly in the tests themself

@williamlardier is also working on refactoring and simplifying setup, please check with him best way to make the changes in some way that will be least complicated to merge.

File: tests/zenko_tests/e2e_config/locations.py [hunk 2]
```diff
@@ -1,22 +1,98 @@
...
+    try:
+        iam_client = Session(
+            aws_access_key_id=account_creds["AccessKeyId"],
+            aws_secret_access_key=account_creds["SecretAccessKey"],
+            aws_session_token=account_creds["SessionToken"],
+        ).client('iam', endpoint_url=IAM_ENDPOINT, region_name='us-east-1')
+
+        user = iam_client.create_user(UserName="crr-user")
+        credentials = iam_client.create_access_key(UserName="crr-user")
+
```
Comment (francoisferrand) [inline tests/zenko_tests/e2e_config/locations.py:27]: could be done directly in tests, we have the API there as well?

File: tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        next => srcUtil.compareObjectsCRR(
+            srcBucket,
+            destUtil,
+            destBucket,
+            key,
+            'customKey',
+            next,
+        ),
+    ], done));
+
```
Comment (francoisferrand) [inline tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js:94]: missing test cases for "null version" object (created before versioning is enabled) : i..e writing version on top of it...

File: tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        ),
+        next => srcUtil.compareACLsCRR(
+            srcBucket,
+            destUtil,
+            destBucket,
+            key,
+            next,
+        ),
+    ], done));
+});
```
Comment (francoisferrand) [inline tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js:277]: missing `it('should not replicate hard delete`)`

File: tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+            destBucket,
+            srcUtil,
+            srcBucket,
+            key,
+            undefined,
+            next,
+        ),
+    ], done));
+
+    it('should replicate tags to source bucket when they get updated on the destination bucket', done => async.series([
```
Comment (francoisferrand) [inline tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js:349]: + same for ACL

File: tests/zenko_tests/node_tests/package.json [hunk 6]
```diff
@@ -39,7 +39,7 @@
     "test_azure_crr": "mocha --tags ${MOCHA_TAGS} --exit -t 10000 --reporter mocha-multi-reporters --reporter-options configFile=config.json  backbeat/tests/crr/azureBackend.js",
     "test_gcp_crr": "mocha --tags ${MOCHA_TAGS} --exit -t 10000 --reporter mocha-multi-reporters --reporter-options configFile=config.json  backbeat/tests/crr/gcpBackend.js",
     "test_one_to_many": "mocha --tags ${MOCHA_TAGS} --exit -t 10000 --reporter mocha-multi-reporters --reporter-options configFile=config.json backbeat/tests/crr/oneToMany.js",
-    "test_crr": "npm-run-all test_aws_crr test_azure_crr test_gcp_crr test_one_to_many",
+    "test_crr": "mocha --tags ${MOCHA_TAGS} --exit -t 10000 --reporter mocha-multi-reporters --reporter-options configFile=config.json backbeat/tests/crr/crr.js",
```
Comment (francoisferrand) [inline tests/zenko_tests/node_tests/package.json:42]: what about the earlier tests: test_aws_crr test_azure_crr test_gcp_crr test_one_to_many

shoud we not run them as well?

File: tests/zenko_tests/e2e_config/accounts.py (hunk 1 above)
Comment (welansari) [reply tests/zenko_tests/e2e_config/accounts.py:54]: The accounts created here are used for creating the CRR users, so they need to be setup before the tests and before we create the CRR locations

File: tests/zenko_tests/e2e_config/locations.py (hunk 2 above)
Comment (welansari) [reply tests/zenko_tests/e2e_config/locations.py:27]: These are needed before the CRR locations are created, i want to avoid creating a location in the tests for stability purposes.

File: tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js (hunk 3 above)
Comment (welansari) [reply tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js:94]: This is an edge case that's known to not work, we have an open ticket about it CLDSRV-632

File: tests/zenko_tests/node_tests/package.json (hunk 6 above)
Comment (welansari) [reply tests/zenko_tests/node_tests/package.json:42]: Those tests were never executed, if i remember correctly i think you had a ticket to investigate why they fail a while ago but we never enabled them

File: tests/zenko_tests/e2e_config/accounts.py (hunk 1 above)
Comment (ghost) [reply tests/zenko_tests/e2e_config/accounts.py:54]: It's fine to do it here, I will migrate this too. I think we have no other choice than this approach for now

File: tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js (hunk 3 above)
Comment (francoisferrand) [reply tests/zenko_tests/node_tests/backbeat/tests/crr/crr.js:94]: can we add a placeholder test already (skipped and referencing the ticket) ?

**Review by francoisferrand — APPROVED** [review]

**Review by ghost — APPROVED** [review]
